drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/popover/ErrorReportPopoverPresenter.java

```diff
@@ -16,51 +16,62 @@
 
 package org.drools.workbench.screens.scenariosimulation.client.popover;
 
+import javax.annotation.PostConstruct;
 import javax.enterprise.context.ApplicationScoped;
 import javax.inject.Inject;
 
-import org.drools.workbench.screens.scenariosimulation.client.utils.ViewsProvider;
 import org.uberfire.mvp.Command;
 
 @ApplicationScoped
 public class ErrorReportPopoverPresenter implements ErrorReportPopover.Presenter {
 
     @Inject
-    protected ViewsProvider viewsProvider;
+    protected ErrorReportPopoverView errorReportPopupView;
```

Comment:
@gitgabrio With this change, we could remove `errorReportPopupView` from `ViewsProvider` class, because no more used.
I was wondering about that change, probably you did it because it's marked as "@ApplicaitonScoped", right? Or is it just to avoid multiple `.getErrorReportPopupView()` calls?

Reply:
@yesamer 
You're right. I'm not sure why ViewsProvider contains `ErrorReportPopoverView getErrorReportPopoverView()`: since `ErrorReportPopoverView` is ApplicationScoped, there is no need. I'll remove.